From def85b3ec41da471b9490ba96dca55f904c34fae Mon Sep 17 00:00:00 2001
From: Steve Kondik <steve@cyngn.com>
Date: Tue, 8 Nov 2016 12:51:59 -0800
Subject: [PATCH 16/19] build: Require devices to opt-in for SDCLANG

 * Only devices which see a measureable benefit should use this
   since it requires more resources to build.
 * Set TARGET_USE_SDCLANG to true in your BoardConfig if your
   device benefits from it.

Change-Id: I12ee20b0627ff0ca121d653cde482e5449a768cc
Signed-off-by: Joe Maples <joe@frap129.org>
---
 core/binary.mk | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/core/binary.mk b/core/binary.mk
index c435cec71..50172a075 100644
--- a/core/binary.mk
+++ b/core/binary.mk
@@ -389,6 +389,17 @@ ifeq ($(my_sdclang),true)
     ifeq ($(my_sdclang_2),true)
         $(error LOCAL_SDCLANG and LOCAL_SDCLANG_2 can not be set to true at the same time!)
     endif
+    ifneq ($(TARGET_USE_SDCLANG),true)
+        my_sdclang := false
+    endif
+endif
+
+ifeq ($(SDCLANG),true)
+    ifeq ($(my_sdclang),)
+        ifeq ($(TARGET_USE_SDCLANG),true)
+            my_sdclang := true
+        endif
+    endif
 endif
 
 ifeq ($(LOCAL_C_STD),)
-- 
2.19.1

